src/promptflow/promptflow/_sdk/_serving/extension/azureml_extension.py

```diff
@@ -175,5 +182,21 @@ def _initialize_connection_provider(self):
                         self.deployment_name = match_result.group(5)
                 else:
                     # raise exception if not found any valid connection provider setting
-                    raise MissingConnectionProvider(message="Missing connection provider, please check whether 'PROMPTFLOW_CONNECTION_PROVIDER' is in your environment variable list.")  # noqa: E501
-            self.connection_provider = AML_CONNECTION_PROVIDER_TEMPLATE.format(self.subscription_id, self.resource_group, self.workspace_name)  # noqa: E501
+                    raise MissingConnectionProvider(
+                        message="Missing connection provider, please check whether 'PROMPTFLOW_CONNECTION_PROVIDER' "
+                        "is in your environment variable list."
+                    )  # noqa: E501
+            self.connection_provider = AML_CONNECTION_PROVIDER_TEMPLATE.format(
+                self.subscription_id, self.resource_group, self.workspace_name
+            )  # noqa: E501
+
+
+def _get_managed_identity_credential_with_retry(**kwargs):
+    from azure.identity import ManagedIdentityCredential
+
+    class ManagedIdentityCredentialWithRetry(ManagedIdentityCredential):
+        @retry(Exception)
```

Comment:
what's the default retry number and machanism?

Reply:
default retry count is 4, and retry intervals are 3, 6, 12. (delay(3) * backoff(2)^retry_time)